src/taskgraph/docker.py

```diff
@@ -237,3 +239,59 @@ def download_and_modify_image():
         raise Exception("No repositories file found!")
 
     return info
+
+
+def load_task(task_id, remove=False):
+    task_def = get_task_definition(task_id)
+
+    if (
+        impl := task_def.get("tags", {}).get("worker-implementation")
+    ) != "docker-worker":
+        print(f"Tasks with worker-implementation '{impl}' are not supported!")
+        return 1
+
+    command = task_def["payload"].get("command")
+    if not command or not command[0].endswith("run-task"):
+        print("Only tasks using `run-task` are supported!")
+        return 1
+
+    # Remove the payload section of the task's command. This way run-task will
+    # set up the task (clone repos, download fetches, etc) but won't actually
+    # start the core of the task. Instead we'll drop the user into a command
+    # shell.
+    if (index := command.index("--")) > -1:
+        command[index + 1 :] = ["echo 'Task setup complete!'"]
```

Comment:
Would it be useful to stash the rest of the command in an env var or something?  So that you could e.g. `eval $ORIGINAL_COMMAND` from the shell.

Reply:
Good idea!